### ISSUE TITLE
Support for google_service_networking_peered_dns_domain

### ISSUE THREAD

**Issue by drebes** [hashicorp/terraform-provider-google#9592]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Service Networking supports the producer network to consume DNS configuration from the consumer network via the PeeredDnsDomain API object. This is useful for products that initiate connections from the producer network into the consumer network, like Apigee, so that FQDNs can be used. This resource is currently missing from Terraform.

### New or Affected Resource(s)

*google_service_networking_peered_dns_domain

### Potential Terraform Configuration

```tf
resource "google_service_networking_peered_dns_domain" "example" {
  name       = "example-com" 
  dns_suffix = "example.com"
  network    = google_compute_network.peering_network.id
  service    = "servicenetworking.googleapis.com" # This should be the default service
}
```

### References

* [gcloud services peered-dns-domains create](https://cloud.google.com/sdk/gcloud/reference/services/peered-dns-domains/create)
* [API reference](https://cloud.google.com/service-infrastructure/docs/service-networking/reference/rest/v1/services.projects.global.networks.peeredDnsDomains/create)

**Comment by DeviaVir** [comment]
part of https://github.com/GoogleCloudPlatform/magic-modules/pull/5158

**Comment by drebes** [comment]
Fixed by GoogleCloudPlatform/magic-modules#5158
